ampd/src/health_check.rs

```diff
@@ -0,0 +1,105 @@
+use error_stack::{Result, ResultExt};
+use std::net::SocketAddr;
+use thiserror::Error;
+use tracing::info;
+
+use axum::{http::StatusCode, routing::get, Json, Router};
+use serde::{Deserialize, Serialize};
+use tokio_util::sync::CancellationToken;
+
+pub struct Server {
+    listener: tokio::net::TcpListener,
+}
+
+#[derive(Error, Debug)]
+pub enum HealthCheckError {
+    #[error("Health check server error: {0}")]
+    Error(String),
+}
+
+impl Server {
+    pub async fn new(bind_addr: SocketAddr) -> Result<Self, HealthCheckError> {
+        Ok(Self {
+            listener: tokio::net::TcpListener::bind(bind_addr)
+                .await
+                .change_context(HealthCheckError::Error(format!(
+                    "Failed binding to addr: {}",
+                    bind_addr
+                )))?,
+        })
+    }
+
+    #[allow(dead_code)]
```

Comment:
The `#[cfg(test)]` attribute is preferred over `#[allow(dead_code)]` for functions only used in tests, as it ensures the item will only be compiled for test builds.

```suggestion
    #[cfg(test)]
```

Reply:
great idea.